### PR TITLE
TR-1436: Fullscreen layout for prototype fullscreen code editor

### PR DESCRIPTION
**Pull request by bkemper**
This is a subtask [TR-1436](https://jira.int.messagesystems.com/browse/TR-1436) of [TR-1204](https://jira.int.messagesystems.com/browse/TR-1204).

### What Changed

- Replaced routing by content part for sub navigation, Contents and Settings (eventually, AB Testing)
- New layout

Screen Shot 2019-05-24 at 11 24 51 AM

### How To Test
- Confirm routing to subnav works (e.g. /templates/edit/recipient-validation-beta-email/next/settings)
- Confirm save button still works

### CODE REVIEW

**Comment by bkemper** [comment]
@rajumsys This PR is ready for review.

File: src/components/layout/Fullscreen.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import WindowSize from 'src/context/WindowSize';
+import styles from './Layout.module.scss';
+
+const Fullscreen = ({ children }) => (
+  <WindowSize>
+    <div className={`${styles.wrapper} ${styles.fullscreen}`}>
+      <main role="main" className={styles.content}>
+        {children}
+      </main>
+      <ScrollToTop/>
```
Comment (rajumsys) [inline src/components/layout/Fullscreen.js]: do we need that? in which case it won't be at top already?

**Comment by bkemper** [comment]
> Text preview shows undefined sometime. Don't step specific steps but might be until we type something.

I fixed this in #1034

**Comment by bkemper** [comment]
> On a smaller screen (try resizing), preview section is at top and editor at bottom. Is that an intentional choice?

Yes, Lynn originally didn't want to show the editor at all on smaller screens, but for now, it will just drop below the preview.

**Comment by bkemper** [comment]
> sections' height not same

Hmm... It is the Tabs vs Controls.  I'll fix that.

File: src/components/fullPage/FullPage.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { ChevronLeft } from '@sparkpost/matchbox-icons';
+import propTypes from 'prop-types';
+import styles from './FullPage.module.scss';
+
+// uses the entire viewport
+const FullPage = ({ breadcrumbRedirectsTo, children, primaryArea, title }) => (
+    <>
+      <div className={styles.FullPageNav}>
+        <div className={styles.FullPageHeaderArea}>
+          <UnstyledLink
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/components/fullPage/FullPage.js]: You are right.  I missed that.  Added... 50768a41f

File: src/components/fullPage/FullPage.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            children={<ChevronLeft size={28} />}
+            className={styles.FullPageBreadcrumb}
+            to={breadcrumbRedirectsTo}
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/components/fullPage/FullPage.js]: Fixed... 23fdad2d1

File: src/components/layout/Fullscreen.js (hunk 1 above)
Comment (bkemper) [reply src/components/layout/Fullscreen.js]: Removed... bf500bf50

**Comment by bkemper** [comment]
Heights should match now... 3c0e61076

**Comment by bkemper** [comment]
@rajumsys This is ready for final review.

**Pull request merged**